presto-main/src/main/java/com/facebook/presto/server/QueryStateInfoResource.java

```diff
@@ -78,4 +80,17 @@ public QueryStateInfoResource(
                         queryManager.getQueryResourceGroup(queryInfo.getQueryId()).map(ResourceGroupId::getRoot).map(rootResourceGroupInfos::get)))
                 .collect(toImmutableList());
     }
+
+    @GET
+    @Path("queryId/{queryId}")
```

Comment:
Why is this a path param and the other is a query param? Let's make them both path params.